ingress/controllers/gce/controller/cluster_manager.go

```diff
@@ -241,24 +241,28 @@ func NewClusterManager(
 	defaultBackendNodePort int64,
 	defaultHealthCheckPath string) (*ClusterManager, error) {
 
-	var config *os.File
-	var err error
+	// TODO: Make this more resilient. Currently we create the cloud client
+	// and pass it through to all the pools. This makes unittesting easier.
+	// However if the cloud client suddenly fails, we should try to re-create it
+	// and continue.
+	var cloud *gce.GCECloud
 	if configFilePath != "" {
 		glog.Infof("Reading config from path %v", configFilePath)
-		config, err = os.Open(configFilePath)
+		config, err := os.Open(configFilePath)
 		if err != nil {
 			return nil, err
 		}
 		defer config.Close()
+		cloud = getGCEClient(config)
+		glog.Infof("Successfully loaded cloudprovider using config %q", configFilePath)
+	} else {
+		// While you might be tempted to refactor so we simply assing nil to the
+		// config and only invoke getGCEClient once, that will not do the right
+		// thing because a nil check against an interface isn't true in golang.
+		cloud = getGCEClient(nil)
```

Comment:
Yeah, it needs to be nil when running this as a pod in the cluster (vs with a valid gce config on the master)